R/dgp-lib.R

```diff
@@ -14,104 +14,61 @@
 #'   
 #' @details If \code{add_err = TRUE}, data is generated from the following 
 #' additive model: 
-#' \deqn{y = y_fun(X, ...) + err_fun(X, y_fun(X), ...), where X = x_fun(...).}
+#' \deqn{y = y_fun(X, ...) + err_fun(X, y_fun(X), ...), where X = X_fun(...).}
 #' 
 #' If \code{add_err = FALSE}, data is generated via:
-#' \deqn{y = err_fun(X, y_fun(X, ...), ...), where X = x_fun(...).}
+#' \deqn{y = err_fun(X, y_fun(X, ...), ...), where X = X_fun(...).}
 #' 
 #' Note that while \code{err_fun()} is allowed to depend on both X and y, it is
 #' not necessary that \code{err_fun()} depend on X or y.
 #' 
-#' If x_fun, y_fun, and err_fun have arguments of the same name, then use the
-#' prefix ".x_", ".y_", ".err_" in front of the argument name (passed via ...) 
-#' to differentiate it for use in x_fun, y_fun, or err_fun, respectively. 
-#' See the examples.
-#'
 #' @examples 
 #' # generate X = 100 x 10 standard Gaussian, y = linear regression model
-#' sim_data <- xy_dgp_constructor(x_fun = MASS::mvrnorm, 
+#' sim_data <- xy_dgp_constructor(X_fun = MASS::mvrnorm, 
 #'                                y_fun = generate_y_linear,
 #'                                err_fun = rnorm, data_split = TRUE,
 #'                                # shared dgp arguments
 #'                                n = 100,
-#'                                # arguments specifically for x_fun
+#'                                # arguments specifically for X_fun
 #'                                .x_mu = rep(0, 10), .x_Sigma = diag(10),
 #'                                # arguments specifically for y_fun
 #'                                .y_betas = rnorm(10), .y_return_support = TRUE,
 #'                                # arguments specifically for err_fun
 #'                                .err_sd = 1)
-#' # or alternatively, (since arguments of x_fun, y_fun, err_fun are unique, 
+#' # or alternatively, (since arguments of X_fun, y_fun, err_fun are unique, 
 #' # with the exception of `n`)
-#' sim_data <- xy_dgp_constructor(x_fun = MASS::mvrnorm, 
+#' sim_data <- xy_dgp_constructor(X_fun = MASS::mvrnorm, 
 #'                                y_fun = generate_y_linear,
 #'                                err_fun = rnorm, data_split = TRUE,
 #'                                # shared dgp arguments
 #'                                n = 100,
-#'                                # arguments specifically for x_fun
+#'                                # arguments specifically for X_fun
 #'                                mu = rep(0, 10), Sigma = diag(10),
 #'                                # arguments specifically for y_fun
 #'                                betas = rnorm(10), return_support = TRUE,
 #'                                # arguments specifically for err_fun
 #'                                sd = 1)
 #'
 #' @export
-xy_dgp_constructor <- function(x_fun, y_fun, err_fun = NULL, add_err = TRUE,
+xy_dgp_constructor <- function(X_fun, y_fun, err_fun = NULL, add_err = TRUE,
```

Comment:
This is more of a question for my understanding than anything else. But suppose I set `y_fun = generate_y_linear()` and want to generate y from `betas` that I generate via a function (e.g., `rnorm`). Would both code snippets below produce the same output (under the same random seed)?

```
xy_dgp_constructor(x_fun = MASS::mvrnorm, y_fun = generate_y_linear, 
                                   n = 100, mu = rep(0, 10), Sigma = diag(10), 
                                   betas = rnorm, .betas_sd = 2)
```

```
xy_dgp_constructor(x_fun = MASS::mvrnorm, y_fun = generate_y_linear, 
                                   n = 100, mu = rep(0, 10), Sigma = diag(10), 
                                   betas = rnorm, .y_.betas_sd = 2)
```

```
xy_dgp_constructor(x_fun = MASS::mvrnorm, y_fun = generate_y_linear, 
                                   n = 100, mu = rep(0, 10), Sigma = diag(10), 
                                   betas = rnorm, .y_sd = 2)
```

Reply:
Good question; will test this out. Probably we should clearly tell users to avoid nested `.` args as much as possible, but this does seem like a legit use case.